g11n-ws/modules/md-service-i18n-l3/src/main/java/com/vmware/vip/core/messages/utils/PseudoMessagesUtils.java

```diff
@@ -47,23 +48,22 @@ public static SingleComponentDTO getPseudoMessages(
 	}
 
 	@SuppressWarnings({ "unchecked", "rawtypes" })
-	public static TranslationDTO getPseudoMessages2(
-			TranslationDTO translationDTO, PseudoConfig pseudoConfig) {
-		TranslationDTO t = new TranslationDTO();
-		BeanUtils.copyProperties(translationDTO, t);
-		JSONArray pseudoList = new JSONArray();
-		JSONArray ja = translationDTO.getBundles();
-		for (Object ob : ja) {
-			if (pseudoConfig.isEnabled()) {
-				JSONObject jo = (JSONObject) ob;
-				Map messages = (Map) jo.get(ConstantsKeys.MESSAGES);
-				jo.put(ConstantsKeys.MESSAGES,
-						JSONUtils.getOrderedMapForPseudo(messages,
-								pseudoConfig.getExistSourceTag()));
-				pseudoList.add(jo);
-			}
-		}
-		t.setBundles(pseudoList);
-		return t;
-	}
+    public static TranslationDTO getPseudoMessages2(
+            TranslationDTO translationDTO, PseudoConfig pseudoConfig) {
+        TranslationDTO t = SerializationUtils.clone(translationDTO);
```

Comment:
BeanUtils.copyProperties(translationDTO, t); 
Why doesn't this work?

Reply:
It is not a deep copy, so when add the pseudo tag, it will change cache latest source translation